### PR TITLE
Issue #27014:new return authorizations by item display

### PR DESCRIPTION
**Pull request by garyhgohoos**

### CODE REVIEW

File: foundation-database/public/tables/metasql/returnAuthorizationItems-detail.mql [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                   JOIN uom ON (uom_id=raitem_qty_uom_id)
+               <? else ?> 
+                   LEFT OUTER JOIN raitem ON (raitem_rahead_id=rahead_id) 
+                   LEFT OUTER JOIN itemsite ON (raitem_itemsite_id=itemsite_id) 
+                   LEFT OUTER JOIN whsinfo ON (itemsite_warehous_id=warehous_id) 
+                   LEFT OUTER JOIN uom ON (uom_id=raitem_qty_uom_id)
+               <? endif ?> 
+ WHERE ((true)
+<? if exists("openOnly") ?> 
+   AND ((raitem_status = 'O') OR (raitem_status IS NULL))
```
Comment (gilmoskowitz) [inline foundation-database/public/tables/metasql/returnAuthorizationItems-detail.mql]: `raitem_status` cannot be null according to [raitem.sql](https://github.com/xtuple/private-extensions/blob/4_10_x/source/inventory/foundation-database/frozen/tables/raitem.sql#L31). Is this not true in practice?

File: foundation-database/public/tables/metasql/returnAuthorizationItems-detail.mql [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+         'error'
+       END AS rahead_expiredate_qtforegroundrole
+  FROM rahead LEFT OUTER JOIN custinfo ON (rahead_cust_id=cust_id)
+               <? if exists("selectedSites") ?> 
+                   JOIN raitem ON (raitem_rahead_id=rahead_id) 
+                   JOIN itemsite ON (raitem_itemsite_id=itemsite_id) 
+                   JOIN site() ON (itemsite_warehous_id=warehous_id) 
+                   JOIN uom ON (uom_id=raitem_qty_uom_id)
+               <? else ?> 
+                   LEFT OUTER JOIN raitem ON (raitem_rahead_id=rahead_id)
```
Comment (gilmoskowitz) [inline foundation-database/public/tables/metasql/returnAuthorizationItems-detail.mql]: The point is to find Items used by R/A's, which would imply that we should always _inner_ join with `raitem`.

**Comment by gilmoskowitz** [comment]
This code really belongs in [private-extensions](https://github.com/xtuple/private-extensions/tree/4_10_x/source/inventory).

**Comment by garyhgohoos** [comment]
i did not move to private-extensions because there are a bunch of "Return Authorization" related files in this repo

**Pull request merged**